### PR TITLE
Preliminary: roslyn view compilation without orchard.web dependencies.

### PR DESCRIPTION
**Pull request by jtkech**
@sebastienros 

- View compilation: So, **just to show my code but wait because still an issue**. E.g `/admin` works, `/orchard.demo/home/display/x` works, `/home/index` works ... But `/orchard.themes/admin/index` fails, need to investigate but here seems not to be related to a missing assembly.

- That said, because we use `dotnet cli api`, some methods relies on the presence of the extensions `project.json` files (as before with e.g `CreateContextForEachFramework()`). So, it seems that we would need these files when Publishing.

- Other methods need a configuration parameter which is `Debug` by default if not provided (as before with e.g `CreateLoadContext()` and now with `CreateExporter()`). So, i think `Debug folders` are parsed and it seems that we need a way to select the right configuration, e.g `Release / Production`.

- **When publishing**, i don't know how we want the files to be organized, in which folders ... E.g when i do dotnet publish in orchard.web folder, dlls are outputed in the same publish folder where i can also see the nuget dlls. So, not sure, but maybe an assembly loading may fail if the dll is already there ...

- **Vscode**: Because now orchard.web project doesn't have all dependencies, i've created a fake project / commands to compile all the project.

Best.

### CODE REVIEW

File: src/Orchard.Hosting.Web/Extensions/ApplicationBuilderExtensions.cs [hunk 1]
```diff
@@ -32,14 +37,49 @@ public static class ApplicationBuilderExtensions
...
+                }
+            }
+
             foreach (var extension in extensionManager.AvailableExtensions())
             {
                 var extensionEntry = extensionManager.LoadExtension(extension);
                 applicationPartManager.ApplicationParts.Add(new AssemblyPart(extensionEntry.Assembly));
-            }
+
+                if (!extensionEntry.Assembly.FullName.StartsWith("Orchard.Core")) {
```
Comment (sebastienros) [inline src/Orchard.Hosting.Web/Extensions/ApplicationBuilderExtensions.cs]: I think we should get rid of this module, it's a lot of pain. We would not support multiple modules inside a single assembly, but that's fine as this is the only one we do it for. If a module is necessary, then like Orchard.Displaymanagement it should just not be a module, but a referenced assembly.

Comment (Jetski5822) [inline src/Orchard.Hosting.Web/Extensions/ApplicationBuilderExtensions.cs]: I think we should get rid of Orchard.Core too... it bread to many cross referenced components in O1... so we should split it for O2

Comment (sebastienros) [inline src/Orchard.Hosting.Web/Extensions/ApplicationBuilderExtensions.cs]: And it makes me sad to create custom code just for this module.

**Comment by jtkech** [comment]
@sebastienros 

PR updated with an `IExtensionLibraryService` for testing ...

It was working fine but after a rebase to solve conflicts and an update of the .NET Core SDK, it fails`:

    "Method not found: 'Microsoft.Extensions.PlatformAbstractions.IRuntimeEnvironment
    Microsoft.Extensions.PlatformAbstractions.PlatformServices.get_Runtime()'."`

So i've re-tried after reverting my commits, and it also fails:

    System.MissingMethodException

So, maybe a nuget package update problem, some of mine have a higher version. Maybe worth to test on your environment before / after my PR. I'll see tomorrow.

Best.

**Comment by alexbocharov** [comment]
@jtkech I see the problem in the package dependencies. I'll update dependencies

**Comment by alexbocharov** [comment]
> It was working fine but after a rebase to solve conflicts and an update of the .NET Core SDK, it fails`:
"Method not found: 'Microsoft.Extensions.PlatformAbstractions.IRuntimeEnvironment
Microsoft.Extensions.PlatformAbstractions.PlatformServices.get_Runtime()'."`
So i've re-tried after reverting my commits, and it also fails:
System.MissingMethodException

Errors should not be https://github.com/OrchardCMS/Orchard2/pull/164

**Comment by jtkech** [comment]
@alexbocharov

Thanks after updating the versions on my local environment all works fine.

@sebastienros 

So, for testing this PR we need the package versions updates.

And before testing you also need to delete the orchard.web bin folder. Runtime loaded modules/themes dlls should not be in the orchard.web bin folder.

Best.

File: src/Projects/project.json [hunk 2]
```diff
@@ -0,0 +1,69 @@
+{
```
Comment (sebastienros) [inline src/Projects/project.json]: What is this file for ?

**Comment by jtkech** [comment]
@sebastienros it's only to be able to compile all the projects (except orchard.web) through a vscode command. So, this file is as a solution file.

**Comment by jtkech** [comment]
@sebastienros @Jetski5822 

So, i've rebased the PR to use the right packages versions, then seems to work fine.

- It relies on the presences of `project.json` files in modules / themes folders.
- Before compiling, delete the orchard.web bin folder, modules / themes dlls should not be there.
- In 2 places i use the `Debug` configuration, so the program looks in `bin/Debug/netstandard1.5` folders. The pattern is `bin/{configuration}/{framework}`. Framework is in project.json but configuration is at compile time (`Debug` by default but can be any name).
- But i think we can specify output path explicitly (and per configuration) in project.json. So, here we need to know which strategy we want to use and how select at runtime the configuration.

Best.

**Pull request merged**